src/Marten/Exceptions/ApplyEventException.cs

```diff
@@ -6,7 +6,7 @@ namespace Marten.Exceptions;
 public class ApplyEventException: MartenException
 {
     public ApplyEventException(IEvent @event, Exception innerException): base(
-        $"Failure to apply event #{@event.Sequence} ({@event.Data}.)", innerException)
+        $"Failure to apply event #{@event.Sequence} Id({@event.Id})", innerException)
```

Comment:
I guess you're right, but I think its better to not have the best log message thinkable, then to have GPDR requirements broken. So maybe this logging can be improved in a future fix? 

I could add the EventTypeName and StreamKey ?? StreamId

Reply:
Currently, we don't know what is GDPR related data or some sensitive data which shouldn't be logged. May be once there is support for GDPR in Marten, there could be a attribute or something which can be applied on a property or field and it can be redacted while logging as well. Until then we can't have a clear cut fix.

Reply:
Well, we could not include Data and include other descriptive properties untill then? That will prevent sensitive data to be unintentially written, especially using records. Which is becoming the default for events as of there ease of use for the case.

Reply:
We will see what could be a short term solution, will discuss with @jeremydmiller and @oskardudycz on this.

Reply:
That event.Id is Marten assigned and I don't think would refer directly to any kind of PII

Reply:
> That event.Id is Marten assigned and I don't think would refer directly to any kind of PII

Just having `event.Id` wouldn't give much details about the exception for troubleshooting purposes.

Reply:
I agree @mysticmind , maybe we can add the event type and event stream id in another PR